hadoop-common-project/hadoop-common/src/main/resources/core-default.xml

```diff
@@ -1522,6 +1522,19 @@
     </description>
 </property>
 
+<property>
+  <name>fs.s3a.metadatastore.fail.on.write.error</name>
```

Comment:
We should use authoritative mode instead of adding a new protperty for this.
There's no case when you want this off when auth mode is on, and on if auth mode is off.

Reply:
I've been looking at it a little differently.  Primarily, I've been struggling with whether anyone would really _ever_ want this setting off (authoritative mode or not).  Why bother using S3Guard if you're ok with the idea that it won't even necessarily prevent seeing list-after-create inconsistencies?  You're effectively getting the same behavior as if you didn't use S3Guard - most of the time you see all the files and sometimes you don't.

I considered crafting this patch without there even being a configuration for this or with the default to true, but figured current behavior must have been intentional and there is an argument I'm not seeing.  Can you guys help me see the side of this where someone would want this false?

Reply:
Ben - I think you are right. That "swallow the failures" probably follows on from the strategy of handling failures in the delete fake directories phase, where we don't want a failure there to escalate. Here we do as it is a sign that the store has gone inconsistent.

If we mandate that update operations always fail the job, well, at least you get to find out when something has gone wrong. In which case: no need for an option, no need for extra complications in testing.

*Is there anyone watching this who disagrees? If so, please justify*